src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -469,4 +490,72 @@ TEST_CASE("request_rate_streaming: test that streaming-specific logic works")
   }
 }
 
+TEST_CASE(
+    "custom_json_data: Check custom json data to ensure that it is processed "
+    "correctly")
+{
+  PerfAnalyzerParameters params{};
+  TestRequestRateManager trrm{params};
+
+  std::shared_ptr<MockModelParser> mmp{
+      std::make_shared<MockModelParser>(false, false)};
+  ModelTensor model_tensor{};
+  model_tensor.datatype_ = "INT32";
+  model_tensor.is_optional_ = false;
+  model_tensor.is_shape_tensor_ = false;
+  model_tensor.name_ = "INPUT0";
+  model_tensor.shape_ = {1};
+  mmp->inputs_ = std::make_shared<ModelTensorMap>();
+  (*mmp->inputs_)[model_tensor.name_] = model_tensor;
+
+  std::shared_ptr<MockDataLoader> mdl{std::make_shared<MockDataLoader>()};
+  const std::string json_str{R"(
+{
+  "data": [
+    {
+      "INPUT0": [2000000000]
+    },
+    {
+      "INPUT0": [2000000001]
+    }
+  ]
+}
+    )"};
+  mdl->ReadDataFromStr(json_str, mmp->Inputs(), mmp->Outputs());
+
+  trrm.parser_ = mmp;
+  trrm.data_loader_ = mdl;
+  trrm.using_json_data_ = true;
+  trrm.execute_ = true;
+  trrm.batch_size_ = 1;
+  trrm.max_threads_ = 1;
+  trrm.schedule_.push_back(std::chrono::microseconds(0));
+  trrm.schedule_.push_back(std::chrono::microseconds(500));
+  trrm.schedule_.push_back(std::chrono::microseconds(1000));
+  trrm.schedule_.push_back(std::chrono::microseconds(1500));
+  trrm.gen_duration_ = std::make_unique<std::chrono::nanoseconds>(1500000);
+  trrm.start_time_ = std::chrono::steady_clock::now();
+
+  std::shared_ptr<TestRequestRateManager::ThreadStat> thread_stat{
+      std::make_shared<TestRequestRateManager::ThreadStat>()};
+  std::shared_ptr<TestRequestRateManager::ThreadConfig> thread_config{
+      std::make_shared<TestRequestRateManager::ThreadConfig>(0, 1)};
+
+  std::future<void> infer_future{std::async(
+      &TestRequestRateManager::Infer, &trrm, thread_stat, thread_config)};
+
+  std::this_thread::sleep_for(std::chrono::microseconds(1250));
+
+  early_exit = true;
+  infer_future.get();
+
+  const auto& recorded_inputs{trrm.stats_->recorded_inputs};
+
+  REQUIRE(trrm.stats_->recorded_inputs.size() >= 4);
+  CHECK(*reinterpret_cast<const int32_t*>(recorded_inputs[0][0]) == 2000000000);
```

Comment:
I think all this casting and pointer conversion should happen above where I mentioned it. This makes assumptions about the size.

Reply:
The reason this is casted to "int32_t" is because higher up in the test, the input datatype is set to `INT32`. So this seems like the appropriate place to cast, since its the same place as the origin of the datatype for the input being cast.